addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/model/LightState.java

```diff
@@ -29,7 +29,7 @@
     private int saturation;
 
     public PercentType getBrightness() {
-        return new PercentType(brightness);
+        return onOff > 0 ? new PercentType(brightness) : PercentType.ZERO;
```

Comment:
The only comment I have is that int and thus this comparison is to me rather surprising for a var called `onOff`

Reply:
The onOff variable is the name of the json variable in the tp-link protocol and contains a value of 0 and (probably only) 1 instead of a boolean value. So not much of a choice in naming and/or content. Here it probably would have been more readable to swap the condition and compare to 0 or just compare to 1 as I see I wrote in the getOnOff method. So I don't know what I was thinking or why I wrote it differently here :man_shrugging: